warp10/src/main/java/io/warp10/standalone/StandaloneAcceleratedStoreClient.java

```diff
@@ -0,0 +1,298 @@
+    protected Boolean initialValue() {
+      return Boolean.FALSE;
+    }
+  };
+
+  private static final ThreadLocal<Boolean> nopersist = new ThreadLocal<Boolean>() {
+    @Override
+    protected Boolean initialValue() {
+      return Boolean.FALSE;
+    }
+  };
+
+  private static final AtomicBoolean instantiated = new AtomicBoolean(false);
+  
+  public StandaloneAcceleratedStoreClient(DirectoryClient dir, StoreClient persistentStore) {
+    
+    instantiated.set(true);
+    
+    this.persistent = persistentStore;
+    this.cache = new StandaloneChunkedMemoryStore(WarpConfig.getProperties(), Warp.getKeyStore());
+
+    this.ephemeral = "true".equals(WarpConfig.getProperty(Configuration.IN_MEMORY_EPHEMERAL)); 
+    
+    //
+    // Preload the cache
+    //
+    
+    long nanos = System.nanoTime();
+    
+    DirectoryRequest request = new DirectoryRequest();
+    request.addToClassSelectors("~.*");
+    Map<String,String> labelselectors = new HashMap<String,String>();
+    labelselectors.put(Constants.APPLICATION_LABEL, "~.*");
+    labelselectors.put(Constants.PRODUCER_LABEL, "~.*");
+    labelselectors.put(Constants.OWNER_LABEL, "~.*");
+    request.addToLabelsSelectors(labelselectors);
+    
+    long end;
+    long start;
+    long n = -1L;
+    
+    if (this.ephemeral) {
+      end = Long.MAX_VALUE;
+      start = Long.MIN_VALUE;
+      n = 1L;
+    } else {
+      end = InMemoryChunkSet.chunkEnd(TimeSource.getTime(), this.cache.getChunkSpan());
+      start = end - this.cache.getChunkCount() * this.cache.getChunkSpan() + 1;
+      n = -1L;
+    }
+    
+    final long now = end;
+    final long then = start;
+    final long count = n;
+    
+    if ("true".equals(WarpConfig.getProperty(Configuration.ACCELERATOR_PRELOAD_ACTIVITY))) {
+      long activityWindow = Long.parseLong(WarpConfig.getProperty(Configuration.INGRESS_ACTIVITY_WINDOW, "-1"));
+      if (activityWindow > 0) {
+        request.setActiveAfter(then / Constants.TIME_UNITS_PER_MS - activityWindow - 1L);
+      }
+    }
+    
+    try {
+      final AtomicLong datapoints = new AtomicLong();
+      
+      MetadataIterator iter = dir.iterator(request);
+      
+      int BATCH_SIZE = Integer.parseInt(WarpConfig.getProperty(Configuration.ACCELERATOR_PRELOAD_BATCHSIZE, "1000"));
+      List<Metadata> batch = new ArrayList<Metadata>(BATCH_SIZE);
+            
+      int nthreads = Integer.parseInt(WarpConfig.getProperty(Configuration.ACCELERATOR_PRELOAD_POOLSIZE, "8"));
+      
+      ThreadPoolExecutor exec = new ThreadPoolExecutor(nthreads, nthreads, 60L, TimeUnit.SECONDS, new LinkedBlockingQueue<Runnable>(nthreads));
+      final AtomicReference<Throwable> error = new AtomicReference<Throwable>();
+      
+      while(iter.hasNext()) {
+        batch.add(iter.next());
+        
+        if (null != error.get()) {
+          throw new RuntimeException("Error populating the accelerator", error.get());
+        }
+        
+        if (BATCH_SIZE == batch.size()) {
```

Comment:
You could  add `|| !iter.hasNext` to avoid adding the `if (!batch.isEmpty())` after.